libs/aws/langchain_aws/embeddings/bedrock.py

```diff
@@ -121,20 +126,36 @@ def validate_environment(self) -> Self:
         return self
 
     def _embedding_func(self, text: str) -> List[float]:
-        """Call out to Bedrock embedding endpoint."""
+        """Call out to Bedrock embedding endpoint with a single text."""
         # replace newlines, which can negatively affect performance.
         text = text.replace(os.linesep, " ")
 
         # format input body for provider
-        provider = self.model_id.split(".")[0]
         input_body: Dict[str, Any] = {}
-        if provider == "cohere":
+
+        if self.provider == "cohere":
             input_body["input_type"] = "search_document"
             input_body["texts"] = [text]
+            response_body = self._invoke_model(input_body)
+            return response_body.get("embeddings")[0]
         else:
             # includes common provider == "amazon"
             input_body["inputText"] = text
+            response_body = self._invoke_model(input_body)
+            return response_body.get("embedding")
+
+    def _cohere_multi_embedding(self, texts: List[str]) -> List[float]:
+        """Call out to Cohere Bedrock embedding endpoint with multiple inputs."""
+        # replace newlines, which can negatively affect performance.
+        texts = [text.replace(os.linesep, " ") for text in texts]
 
+        input_body: Dict[str, Any] = {}
+        input_body["input_type"] = "search_document"
+        input_body["texts"] = texts
```

Comment:
```suggestion
        input_body: Dict[str, Any] = {
            "input_type": "search_document",
            "texts": texts
        }
```

Reply:
Good callout -- made this simpler / terser with: [fb3e3c5](https://github.com/langchain-ai/langchain-aws/pull/350/commits/fb3e3c597ed4dc26392172c5c8fbb318261aa025)